sycl/source/backend.cpp

```diff
@@ -109,6 +135,22 @@ __SYCL_EXPORT queue make_queue(pi_native_handle NativeHandle,
   }
 }
 
+__SYCL_EXPORT queue make_queue2(pi_native_handle NativeHandle,
+                                bool IsImmCmdList, const context &Context,
+                                const device *Device, bool KeepOwnership,
+                                const property_list &PropList,
+                                const async_handler &Handler, backend Backend) {
+  if (Device) {
```

Comment:
I think `Device` should always be non-null on this version 4 of `make_queue`  so you should not need this code and just inline `make_queue_impl2` here

Reply:
Yes, I have removed the Device check.